### PR TITLE
Retrieve contacts, arms, interventions, and study design

### PR DESCRIPTION
**Pull request by ChristineDuong**
Adds in the backup service (from CT.gov):
* Add overall contact (`overall_contact`) and overall backup contact (`overall_backup_contact`) to `ResearchStudy.contacts`
* Add arm groups to `ResearchStudy.arm`
* Add interventions and arm/interventions cross reference in `ResearchStudy.protocol`
* Adds in study design to `ResearchStudy.category`

**Submitter:**
- [X] Make sure test coverage didn’t decrease. If you are allowing the test coverage to drop, leave an explanation as to why:
- [X] Does an update need to be made to the documentation with these changes? **N/A**
- [X] Make sure there is an update to service library reference in the service wrappers/template once this PR is merged.
- [X] Does an update need to be made to the engine? **Yes, there is a following PR**
- [X] Was the new feature tested by unit tests?
- [X] Was the new feature tested by a manual, end-to-end test?

### CODE REVIEW

File: src/fhir-types.ts [hunk 1]
```diff
@@ -160,6 +160,18 @@ export interface Reference {
   display?: string;
 }
 
+export type PublicationStatus = '	draft' | 'active' | 'retired' | 'unknown';
```
Comment (dmpotter44) [inline src/fhir-types.ts]: Is this supposed to be `'	draft'` with the tab? I assume that's a copy-paste error from the FHIR specs.

File: src/clinicalstudy.ts [hunk 2]
```diff
@@ -141,7 +141,7 @@ export type EnrollmentStruct =
 // Arm Group
 
 export interface ArmGroupStruct {
-  arm_group_label: One<string>;
+  arm_group_label?: One<string>;
```
Comment (dmpotter44) [inline src/clinicalstudy.ts]: According to the schema `arm_group_label` is not optional - is it missing in some of the results returned?

**Review by dmpotter44 — COMMENTED** [review]
I have a couple of questions (included as comments).

File: src/fhir-types.ts (hunk 1 above)
Comment (ChristineDuong) [reply src/fhir-types.ts]: Yes this was a typo (from copy-pasting)! I have removed this in d3a48a3.

File: src/clinicalstudy.ts (hunk 2 above)
Comment (ChristineDuong) [reply src/clinicalstudy.ts]: No, you are correct. I have now removed this in d3a48a3

**Review by dmpotter44 — DISMISSED** [review]

**Review by zlister — DISMISSED** [review]

**Review by zlister — APPROVED** [review]

**Pull request merged**